components/product/ProductSlider/ProductSlider.tsx

```diff
@@ -17,8 +18,25 @@ const ProductSlider: FC = ({ children }) => {
     },
   })
 
+  // Stop the history navigation gesture on touch devices
+  useEffect(() => {
+    const preventNavigation = (event: TouchEvent) => {
+      const touchXPosition = event.touches[0].pageX
+      if (touchXPosition > 10 && touchXPosition < window.innerWidth - 10) return
+      event.preventDefault()
+    }
+
+    sliderContainerRef.current!
+      .addEventListener('touchstart', preventNavigation)
+
+    return () => {
+      sliderContainerRef.current!
+      .removeEventListener('touchstart', preventNavigation)
+    }
+  }, [])
+
   return (
-    <div className={s.root}>
+    <div className={s.root} ref={sliderContainerRef}>
```

Comment:
Hey :), asking out of curiosity: why not use the onTouchStart event directly on the div declaratively instead of attaching the listener via ref?

Reply:
Haha good question. Actually there's no particular reason, I'm just used to manage listeners like that :D